### ISSUE TITLE
Intermittent test failures in :service-glean? (only on CI?)

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/android-components#2359]
When 9a998e1b1c370d319f9264a8995171f60846b939 merged to master, the tests failed with ([logs](https://tools.taskcluster.net/groups/JjqQ7sfLTLOBUGX4J2pLQw/tasks/EiK0DdyDSceAQYo_hdsdVQ/runs/0/logs/public%2Flogs%2Flive.log)):
```
> Task :service-glean:testDebugUnitTest
Calling mockable JAR artifact transform to create file: /root/.gradle/caches/transforms-1/files-1.1/android.jar/c8c477ee37fb4f33232e803b5ef60de4/android.jar with input /build/android-sdk/platforms/android-28/android.jar
mozilla.components.service.glean.scheduler.MetricsPingSchedulerTest > glean should close the measurement window for overdue pings before recording new data FAILED
    java.lang.AssertionError
        Caused by: java.lang.AssertionError
[taskcluster:error] Task timeout after 7200 seconds. Force killing container.
[taskcluster 2019-03-14 18:45:29.994Z] === Task Finished ===
[taskcluster 2019-03-14 18:45:29.994Z] Unsuccessful task run with exit code: -1 completed in 7201.478 seconds
```

The commit after that, had no test failures: https://github.com/mozilla-mobile/android-components/commit/2618f26212e74fd705c27a432e418fb94c52d359

I open a PR rebasing on top of both of these commits (#2320) and TaskCluster has failed with test failures in `:service-glean` despite me not changing that module with ([logs](https://tools.taskcluster.net/groups/Hd8LzL3SRJyYCMan4RgC5g/tasks/Q3w-GSAsTt6c59iC29vtUA/runs/0/logs/public%2Flogs%2Flive.log)):
```
...
mozilla.components.service.glean.scheduler.MetricsPingSchedulerTest > collectMetricsPing must correctly trigger the collection of the metrics ping FAILED
    java.lang.AssertionError at MetricsPingSchedulerTest.kt:265
mozilla.components.service.glean.scheduler.MetricsPingSchedulerTest > glean should close the measurement window for overdue pings before recording new data FAILED
    java.lang.AssertionError at MetricsPingSchedulerTest.kt:423
205 tests completed, 13 failed
> Task :service-glean:testDebugUnitTest FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':service-glean:testDebugUnitTest'.
> There were failing tests. See the report at: file:///build/android-components/components/service/glean/build/reports/tests/testDebugUnitTest/index.html
```

(There are more test failures, this is truncated).

I wonder if there are intermittent failures in the `:service-glean` module: I was unable to reproduce locally. Maybe this is only a problem on TaskCluster.

**Comment by mcomella** [comment]
CC @Dexterp37

**Comment by Dexterp37** [comment]
Gah, thanks @mcomella , these are definitely glean related test intermittents, due to 9a998e1. Sorry for the trouble. I'll try to deal with these today.

**Comment by Dexterp37** [comment]
This is being worked on in #2373.

**Comment by Dexterp37** [comment]
@mcomella @pocmo #2373 was merged and addressed a few problems. We also landed improved logging to easier diagnose the remaining issues. I think I'd like to keep this issue open: if you stumble upon any other intermittent madness for glean, would you kindly post the related failure in here?

**Comment by travis79** [comment]
I think we have this fixed now as of #2429

**Comment by Dexterp37** [comment]
Closing this, as we haven't seen new intermittents related to glean recently.
